quipucords/quipucords-ui
# build(webpack): discovery-334 webpack 5

## What's included

- build(webpack): discovery-334 webpack 5

### Notes
- moves from CRA (Create React App) over to a direct Webpack 5 build
- requires NodeJS 18+ now, this may cause issues with the required PR checks
- ~appears as if the remaining Patternfly 3 components need to be removed before this PR can be merged~ see #210 
   - ~pf3 affects the top and left navigation bars~
   - there are minor color style differences that will also affect the login/logout pages


## How to test

> There is now a NODEJS ENGINE CHECK in [`package.json`](https://github.com/quipucords/quipucords-ui/pull/209/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R16) set to NodeJS 18. If you have an earlier version of NodeJS you can attempt to roll that version down for temporary testing, however it is outside of scope to allow earlier versions of NodeJS to work ... [NVM (Node Version Manager used to be used on MacOS to help switch versions)](https://formulae.brew.sh/formula/nvm)
### Coverage and basic unit test check
1. update the NPM packages with `$ yarn`
1. `$ yarn test`
1. confirm tests come back clean

### Interactive unit test check
1. update the NPM packages with `$ yarn`
1. `$ yarn test:dev`
1. confirm local development interactive testing still functions correctly (make a code change and see if it breaks)

### Local run check, mock API
1. update the NPM packages with `$ yarn`
1. make sure Docker/Podman is running
1. `$ yarn start`
1. confirm local run with mock API still functions and code changes update the GUI

### Local run check, API
1. update the NPM packages with `$ yarn`
1. make sure Docker/Podman is running
1. `$ yarn start:stage`
1. confirm local run with API still functions and code changes update the GUI


### Check the build
1. update the NPM packages with `$ yarn`
1. `$ yarn build`
1. confirm the build completes without error


## Example

...

## Updates issue/story

[DISCOVERY-334](https://issues.redhat.com/browse/DISCOVERY-334)
